core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -208,6 +210,7 @@ class ReplicaFetcherThreadTest {
 
     //Assert that truncate to is called exactly once (despite two loops)
     verify(partition, times(3)).truncateTo(anyLong(), anyBoolean())
+    assertTrue(mockBrokerEpochSupplier.getCounter() > 0)
```

Comment:
It is really weird to add those assertions in random tests. They don't bring much value so I would remove them.